ra/ra.go

```diff
@@ -897,16 +897,35 @@ func (ra *RegistrationAuthorityImpl) recheckCAA(ctx context.Context, authzs []*c
 			ch <- err
 		}(authz)
 	}
-	var caaFailures []string
-	for _ = range authzs {
-		if err := <-ch; berrors.Is(err, berrors.CAA) {
-			caaFailures = append(caaFailures, err.Error())
-		} else if err != nil {
-			return err
+	var subErrors []berrors.SubBoulderError
+	var firstBadIdent *identifier.ACMEIdentifier
+	for _, authz := range authzs {
+		err := <-ch
+		if err != nil {
+			if firstBadIdent == nil {
+				firstBadIdent = &authz.Identifier
+			}
+			if bErr, ok := err.(*berrors.BoulderError); ok && berrors.Is(bErr, berrors.CAA) {
+				subErrors = append(subErrors, berrors.SubBoulderError{
+					Identifier:   authz.Identifier,
+					BoulderError: bErr})
+			} else {
+				return err
+			}
 		}
 	}
-	if len(caaFailures) > 0 {
-		return berrors.CAAError("Rechecking CAA: %v", strings.Join(caaFailures, ", "))
+	if len(subErrors) > 0 {
+		var detail string
+		if len(subErrors) == 1 {
+			detail = fmt.Sprintf("Rechecking CAA for %q: %s", firstBadIdent.Value, subErrors[0].BoulderError.Detail)
+		} else {
+			detail = fmt.Sprintf("Rechecking CAA for %q and %d more identifiers failed. "+
```

Comment:
I'm not sure including the first ident here is super helpful, it might actually be slightly confusing. Perhaps we could say `Rechecking CAA failed for multiple identifiers`.

Reply:
I disagree that it isn't helpful. There aren't going to be any ACME clients that implement sub-problem support for some time. I'd prefer to keep at least one identifier in the error message, especially for the initial implementation, so that if a user is using a client that doesn't display subproblem messages they'll still be able to tackle the problem iteratively with the top level error msg.

Reply:
Ah yeah good point. I guess having `Refer to sub-problems for more information` in that case will be slightly confusing to users, but not really anything we can about that. Oh well.